tests/integration/eoa/test_ethaa.py

```diff
@@ -17,118 +17,95 @@
 @pytest.mark.asyncio
 class TestExternallyOwnedAccount:
     class TestComputeStarknetAddress:
+        @pytest.mark.parametrize("address_idx", [i for i in range(0, 4)])
         async def test_should_return_same_address_as_contract(
-            self, externally_owned_account, addresses
+            self, deployer, addresses, address_idx
         ):
-            (
-                deployer,
-                evm_eoas,
-            ) = externally_owned_account
-
-            for i in range(0, len(evm_eoas)):
-                call_info = await deployer.compute_starknet_address(
-                    evm_address=web3.Web3.toInt(hexstr=evm_eoas[i].address)
-                ).call()
-                assert (
-                    call_info.result.contract_address
-                    == addresses[i].starknet_contract.contract_address
-                )
+            address = addresses[address_idx]
+            call_info = await deployer.compute_starknet_address(
+                evm_address=web3.Web3.toInt(hexstr=address.address)
+            ).call()
+            assert (
+                call_info.result.contract_address
+                == address.starknet_contract.contract_address
+            )
 
+        @pytest.mark.parametrize("address_idx", [i for i in range(0, 4)])
         async def test_should_return_the_eth_address_used_at_deploy(
-            self, externally_owned_account, addresses
+            self, deployer, addresses, address_idx
         ):
-            (
-                deployer,
-                evm_eoas,
-            ) = externally_owned_account
-            for i in range(0, len(addresses)):
-                call_info = (
-                    await addresses[i].starknet_contract.get_eth_address().call()
-                )
-                assert call_info.result.eth_address == web3.Web3.toInt(
-                    hexstr=evm_eoas[i].address
-                )
+            address = addresses[address_idx]
+            call_info = await address.starknet_contract.get_eth_address().call()
+            assert call_info.result.eth_address == web3.Web3.toInt(
```

Comment:
it's ok like that but I'd use `int(address.address, 16)` as it's more explicit about the simple transformation used